Helper/Setup.php

```diff
@@ -107,9 +102,7 @@ public function upgradeData(string $mode, ModuleDataSetupInterface $setup, Modul
             $this->customStatus->setCustomStatuses('Running Data '.$mode);
 
             // Dispatch event to NS8 Protect that module has been installed/upgraded
-            if (!$this->scopeConfig->getValue('ns8/protect/token')
-                && !$this->registry->registry(self::ACCESS_TOKEN_SET_KEY)
```

Comment:
I changed this to only look for the presence of `ns8/protect/token` because checking for both seemed redundant.  If `ns8/protect/token` isn't set, things obviously won't work, and we need to try to get+set the value regardless of whether or not `ACCESS_TOKEN_SET_KEY` has been set.

Reply:
@cwalv The set-up in some upgrade instances can run twice without the scopeConfig value being updated which is why the registry call was added here. The first iteration of set-up will set the registry call, then the second iteration which will not fire because the the second part of the IF statement evals to false and then the scopeConfig is saved. This was a work around because Magento wasn't persisting the scopeConfig value save until after the module upgrade.

Reply:
@CTDoriott the redundant check was preventing me from successfully installing because I was in a state where `ns8/protect/token` didn't exist, but the `ACCESS_TOKEN_SET_KEY` value was set.  I think we should probably talk this thru.

Reply:
@cwalv Sounds good! Let me know when you have time and we can hop on a call 👍

Reply:
@cwalv I was able to confirm if there is no `module_setup` record (so Magento believes this is a fresh install) and no protect token in the `core_config_data` then the set-up fails with two attempts to fetch the access token without the registry check present for Magento  2.3.2.

Reply:
@CTDoriott I'll go ahead and remove that part of the change from this PR then.  I do think it'd be worth revisiting, maybe in a separate ticket, what happens from a user's perspective when:

1. The call to `InstallerClient::install('magento', $installRequestData)` fails
 - with this PR, the call to `setup:upgrade` will fail along with it now, but I'm not sure what the user experience would be if installing from the marketplace

2. The registry value is set, but the token is not

3. The token is set, but it's decrypting it fails

I've seen all 3 of these happen in development.

re: #2, I don't know how the registry value was set when the token was not, looking at the code, but running it in the debugger I found that's why it wasn't calling `InstallerClient::install('magento', $installRequestData)` for me initially.

re: #3, Andrew ran into this yesterday, probably due to reinstalling Magento without clearing out the `magento2` database (so the token was created with an old install, and attempting to decrypt it just fails silently).

Reply:
@cwalv Revisiting this later definitely sounds like a good plan 👍.  

The biggest concerns with the registry were #2 & #3 - I think #2 is happening consistently in Magento for me, for #3 I think that is likely the case as well.